src/coord/src/coord.rs

```diff
@@ -576,21 +576,31 @@ where
     ) {
         match cmd {
             Command::Startup { session, tx } => {
-                let mut messages = vec![];
-                let catalog = self.catalog.for_session(&session);
-                if catalog
-                    .resolve_database(catalog.default_database())
-                    .is_err()
-                {
-                    messages.push(StartupMessage::UnknownSessionDatabase);
-                }
                 if let Err(e) = self.catalog.create_temporary_schema(session.conn_id()) {
                     let _ = tx.send(Response {
                         result: Err(anyhow::Error::from(e)),
                         session,
                     });
                     return;
                 }
+
+                let catalog = self.catalog.for_session(&session);
+                if catalog.resolve_role(session.user()).is_err() {
+                    let _ = tx.send(Response {
+                        result: Err(anyhow!("unknown user \"{}\"", session.user())),
```

Comment:
In postgres a valid username is `"`, which would be display as four double quotes. We should perhaps make this user thing a real type that knows how to display itself properly.